### ISSUE TITLE
GPMSA indexing error: simulation vs. experiment

### ISSUE THREAD

**Issue by briadam** [libqueso/queso#553]
In running a Dakota + QUESO/GPMSA problem using dev (76ad824) with 10 simulations and 100 experiment observations (of a scalar response), I ran into an out of bounds memory access.  From context this looks like the simulationScenarios are being indexed where the experiment ones should be.

Offending lines are https://github.com/libqueso/queso/blob/dev/src/gp/src/GPMSA.C#L296 and 297

Possible fix:
```
diff --git a/queso/src/gp/src/GPMSA.C b/queso/src/gp/src/GPMSA.C
index bda5383..624c4d7 100644
--- a/queso/src/gp/src/GPMSA.C
+++ b/queso/src/gp/src/GPMSA.C
@@ -293,8 +293,8 @@ GPMSAEmulator<V, M>::lnValue(const V & domainVector,
       // If we're in the experiment cross correlation part, need extra
       // foo: Sigma_delta/Sigma_v and Sigma_y
       if (i < this->m_numExperiments && j < this->m_numExperiments) {
-        typename SharedPtr<V>::Type cross_scenario1 = (this->m_simulationScenarios)[i];
-        typename SharedPtr<V>::Type cross_scenario2 = (this->m_simulationScenarios)[j];
+        typename SharedPtr<V>::Type cross_scenario1 = (this->m_experimentScenarios)[i];
+        typename SharedPtr<V>::Type cross_scenario2 = (this->m_experimentScenarios)[j];
         prodDiscrepancy = 1.0;
         unsigned int discrepancyCorrStrStart =
           dimParameter + num_svd_terms + dimParameter + dimScenario + 1 +
```

**Comment by briadam** [comment]
@dmcdougall I added this to the v0.57.0, but if that's a scope overreach, feel free to adjust.

**Comment by dmcdougall** [comment]
That change makes sense to me.

**Comment by dmcdougall** [comment]
@briadam No, the milestone is good.  Thanks for doing that.

**Comment by roystgnr** [comment]
Great catch, thanks.

Let me make sure that's the right fix before we commit it, though; I'd hate to change simulation->experiment for consistency's sake and discover that, say, the correct fix was numExperiments->numSimulations instead.

**Comment by briadam** [comment]
Yeah, for sure.  Given the next runtime errors I'm getting from a different context, it deserves considerable scrutiny.

**Comment by dmcdougall** [comment]
@roystgnr or @briadam, I'm happy to give a pull request a third pair of eyes.

**Comment by dmcdougall** [comment]
This was fixed in #554.
